### DISCUSSION TITLE
In which physical location (or region) GitHub is storing the repositories

### DISCUSSION THREAD

**Discussion by hiroto729** [community/community#27606] [Other Feature Feedback, Questions, & Ideas] [answered]
We will use GitHub for storing our code. For compliance reasons, we need to know in which physical location (or region) GitHub store our data.
If it is differ with the country from which we contract, how about in case we contract in U.S.?

**Comment by FahdSeddik** [comment]
Hey!
Main servers are in the US based on my observations. You can find more details from 2017 [here](https://github.blog/2017-10-12-evolution-of-our-data-centers/).

**Comment by MrSarno** [comment]


**Reply by apphoarder** [reply]
This is not quite true.

I work on projects that fall under UK export control.  Some of the code that has been created for these projects is also considered to be a controlled good and cannot be stored outside of the UK without an export licence.  Therefore, we can only use cloud-based services where the servers are UK-based.

**Reply by Vanilagy** [reply]
People disagree with everything nowadays, even empty messages 😭 

![CleanShot 2024-10-14 at 10 05 01@2x](https://github.com/user-attachments/assets/c24e3405-e6b2-442c-bc57-f2d184056336)

**Comment by zioalex** [comment]
Hi all,
any update on this topic?

**Comment by jsoref** [comment, marked as answer]
You're looking for [GitHub Enterprise Cloud with data residency](https://docs.github.com/en/enterprise-cloud@latest/admin/data-residency/about-github-enterprise-cloud-with-data-residency)

> By default, GitHub stores data for GitHub.com in the USA. When you adopt GitHub Enterprise Cloud with data residency, you can choose where your company's code and data are stored. Your enterprise will be hosted on a dedicated subdomain of GHE.com.
>
> The available regions are:
> * The EU
> * Australia
